### PR TITLE
MCOL-5104: Improved algorithm for STD/VAR functions 

### PR DESCRIPTION
**Pull request by mariadb-AndreyPiskunov**
Welford's single-pass algorithm for STDDEV_SAMP, STDDEV_POP, VAR_POP, VAP_SAMP functions and window functions.
(ticket MCOL-5104). Also tests for them.

### CODE REVIEW

**Comment by mariadb-LeonidFedorov** [comment]
@drrtuy take a note, this PR deletes ignore on ARM mtr tests, so we can have it full working

File: utils/windowfunction/wf_stats.h [hunk 1]
```diff
@@ -40,8 +40,8 @@ class WF_stats : public WindowFunctionType
   static boost::shared_ptr<WindowFunctionType> makeFunction(int, const string&, int, WindowFunctionColumn*);
 
  protected:
-  long double fSum1;
-  long double fSum2;
+  long double fMean;
```
Comment (drrtuy) [inline utils/windowfunction/wf_stats.h]: According with the styling guide new attributes must be prefixed with _ thus fM2sum_ is the name.

File: utils/rowgroup/rowaggregation.cpp [hunk 2]
```diff
@@ -4281,18 +4284,39 @@ void RowAggregationUMP2::doAvg(const Row& rowIn, int64_t colIn, int64_t colOut,
...
 {
-  fRow.setDoubleField(fRow.getDoubleField(colOut) + rowIn.getDoubleField(colIn), colOut);
-  fRow.setLongDoubleField(fRow.getLongDoubleField(colAux) + rowIn.getLongDoubleField(colIn + 1), colAux);
-  fRow.setLongDoubleField(fRow.getLongDoubleField(colAux + 1) + rowIn.getLongDoubleField(colIn + 2),
-                          colAux + 1);
+  double count = fRow.getDoubleField(colOut);
+  long double mean = fRow.getLongDoubleField(colAux);
+  long double M2 = fRow.getLongDoubleField(colAux + 1);
+
+  double block_count = rowIn.getDoubleField(colIn);
```
Comment (drrtuy) [inline utils/rowgroup/rowaggregation.cpp]: We use camel case for variable naming.

File: mysql-test/columnstore/basic/r/std_aggregate_columnstore.result [hunk 3]
```diff
@@ -0,0 +1,217 @@
+DROP DATABASE IF EXISTS std_test_db;
```
Comment (drrtuy) [inline mysql-test/columnstore/basic/r/std_aggregate_columnstore.result:1]: The test should be removed b/c it is outside the scope of our control thus any related server change will break the test and we can do nothing with this.

**Review by drrtuy — CHANGES_REQUESTED** [review]
Thx for the patch. It is clean and nice.
Left some comments + discussed the patch offline.

**Review by mariadb-LeonidFedorov — APPROVED** [review]

**Review by drrtuy — APPROVED** [review]

**Pull request merged**